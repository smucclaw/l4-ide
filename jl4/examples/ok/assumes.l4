ASSUME foo IS A FUNCTION FROM NUMBER TO BOOLEAN
#EVAL foo 2

ASSUME bar IS A  BOOLEAN
baz b MEANS IF b THEN 2 ELSE 3
#EVAL baz bar
#EVAL IF bar THEN 2 ELSE 3
