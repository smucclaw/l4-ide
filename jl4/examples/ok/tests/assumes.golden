Parsing successful
Typechecking successful
Evaluation successful
assumes.l4:2:1-12:
  I could not continue evaluating, because I needed to know the value of
    foo
  but it is an assumed term.
assumes.l4:6:1-14:
  I could not continue evaluating, because I needed to know the value of
    bar
  but it is an assumed term.
assumes.l4:7:1-27:
  I could not continue evaluating, because I needed to know the value of
    bar
  but it is an assumed term.
